src/main/java/hudson/plugins/git/BranchSpec.java

```diff
@@ -97,7 +97,11 @@ public boolean matches(String item, EnvVars env) {
     }
 
     private String getExpandedName(EnvVars env) {
-        return env.expand(name);
+        String expandedName = env.expand(name);
+        if (expandedName.length() == 0) {
+            return "**";
```

Comment:
This changes default behaviour from not to do anything for "" to build everything.
Imho it's bad and can break many existed builds.